dDatabase/CDClientDatabase/CDClientTables/CDDeletionRestrictionsTable.cpp

```diff
@@ -0,0 +1,57 @@
+#include "CDDeletionRestrictionsTable.h"
+#include "GeneralUtils.h"
+#include "eDeletionRestrictionsCheckType.h"
+
+CDDeletionRestriction CDDeletionRestrictionsTable::Default = {
+	.id = 0,
+	.restricted = false,
+	.ids = {},
+	.checkType = eDeletionRestrictionsCheckType::MAX
+};
+
+void CDDeletionRestrictionsTable::LoadValuesFromDatabase() {
+
+	uint32_t size = 0;
+	auto tableSize = CDClientDatabase::ExecuteQuery("SELECT COUNT(*) FROM CurrencyTable");
+	while (!tableSize.eof()) {
+		size = tableSize.getIntField(0, 0);
+
+		tableSize.nextRow();
+	}
+
+	tableSize.finalize();
+
+	auto& entries = GetEntriesMutable();
+
+	auto tableData = CDClientDatabase::ExecuteQuery("SELECT * FROM DeletionRestrictions");
+	while (!tableData.eof()) {
+		CDDeletionRestriction entry;
+		entry.id = tableData.getIntField("id", -1);
+		if (entry.id == -1) continue;
+		entry.restricted = tableData.getIntField("restricted", -1);
+		const std::string raw_ids = tableData.getStringField("ids", "");
+		if (!raw_ids.empty()) {
+			for (const auto& idstr : GeneralUtils::SplitString(raw_ids, ',')) {
+				if (!idstr.empty()) {
+					const auto id = GeneralUtils::TryParse<int32_t>(idstr).value_or(-1);
+					if (id != -1) entry.ids.push_back(id);
```

Comment:
```suggestion
					const auto id = GeneralUtils::TryParse<int32_t>(idstr);
					if (id) entry.ids.push_back(id);
```
implicitly covertible to bool.  also .has_value() works too for the check without the 2 comparisons :)